File: backend/spring_api/src/main/java/com/zargo/fivejars/spring_api/features/jars/dtos/CreateJarRequest.java
package com.zargo.fivejars.spring_api.features.jars.dtos;

import jakarta.validation.constraints.*;
import java.math.BigDecimal;

public record CreateJarRequest(
        @NotBlank(message = "Jar name is required")
        @Size(max = 100, message = "Jar name must be under 100 characters")
        String name,

        @Size(max = 255, message = "Description too long")
        String description,

        @NotNull(message = "Coefficient is required")
        @DecimalMin(value = "0.00", message = "Coefficient cannot be negative")
        @DecimalMax(value = "100.00", message = "Coefficient cannot exceed 100%")
        BigDecimal coefficient
) {}

File: flutter_frontend/lib/features/dashboard/dtos/create_jar_request.dart
import 'package:decimal/decimal.dart';

class CreateJarRequest {
  String name;
  String description;
  Decimal coefficient;

  CreateJarRequest({
    required this.coefficient,
    required this.description,
    required this.name,
  });

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'coefficient': coefficient.toString(),
    };
  }
}
